lisa/wa.py

```diff
@@ -645,3 +645,75 @@ def get_stats(self, **kwargs):
             agg_cols=['iteration', 'frame_id'],
             **kwargs,
         )
+
+
+class WASysfsExtractorCollector(WAArtifactCollectorBase):
+    """
+    WA collector for the syfs-extractor augmentation.
+
+    **Example**::
+
+        def pixel6_energy_meter(df):
+            # Keep only CPU's meters
+            df = df[df.value.str.contains('S4M_VDD_CPUCL0|S3M_VDD_CPUCL1|S2M_VDD_CPUCL2')]
+            df[['variable', 'value']] = df.value.str.split(', ', expand=True)
+
+            def _clean_variable(variable):
+                if 'S4M_VDD_CPUCL0' in variable:
+                    return 'little-energy'
+                if 'S3M_VDD_CPUCL1' in variable:
+                    return 'mid-energy'
+                if 'S2M_VDD_CPUCL2' in variable:
+                    return 'big-energy'
+                return ''
+
+            df['variable'] = df['variable'].apply(_clean_variable)
+            df['value'] = df['value'].astype(int)
+            df['unit'] = "bogo-ujoules"
+
+            # Add a total energy variable
+            df = pd.concat([
+                df, pd.DataFrame(data={
+                    'variable': 'total-energy',
+                    'value': [df['value'].sum()]
+                })]
+            )
+            df.ffill(inplace=True)
+
+            return df
+
+        df = WAOutput('.').get_collector('sysfs-extractor',
+                path='/sys/bus/iio/devices/iio:device0/energy_value',
+                df_postprocess=pixel6_energy_meter).df
+    """
+    NAME = 'sysfs-extractor'
+
+    def __init__(self, wa_output, path, type='diff', **kwargs):
+        allowed_types = ['diff', 'before', 'after']
+
+        if type not in allowed_types:
+            ValueError(f'{self.__class__.__qualname__} type must be one of {allowed_types}')
+
+        self._ARTIFACT_NAME = f'{path} [{type}]'
+
+        # TODO: WA's sysfs-extractor augmentation can actually use a path
+        # to collect several files not only one...
+        self._filename = os.path.basename(path)
+
+        super().__init__(wa_output, **kwargs)
+
+    def _get_artifact_df(self, path):
+        # WA given path is actually dirname
+        path = os.path.join(path, self._filename)
+
+        with open(path, 'r') as f:
+            raw_file = f.readlines()
+
+        raw_file = [line.strip() for line in raw_file]
+
+        df = pd.DataFrame(data={
+                'variable': self._filename,
+                'value': raw_file,
+                'unit': 'unknown'
```

Comment:
AFAICT the other collectors use `''` for unknown units. There is also a chance it can work without the column at all (but maybe not, I can't remember)